blog/13-01-2023-build-a-generic-staking-dapp-using-foundry-and-nextjs/build-a-generic-staking-dapp-using-foundry-and-nextjs.md

```diff
@@ -0,0 +1,1837 @@
+---
+title: Build a generic staking dApp using Foundry and Nextjs.
+description: Compiling, testing and deploying contracts using foundry with frontend using Nextjs
+authors:
+  - name: Isaac Jesse
+    title: Web3, Smart Contract Developer
+    url: https://github.com/bobeu
+    image_url: https://github.com/bobeu.png
+tags: [foundry, celo, developers, solidity, react, nextjs]
+hide_table_of_contents: true
+slug: /tutorials/build-a-generic-staking-dapp-using-foundry-and-nextjs
+---
+
+![header](images/1.png)
+
+# Introduction
+
+In the past years, Javascript has been very helpful in web3 development : compiling, testing, deployment and interacting with smart contracts. It was however argued that Hardhat was a significant improvement over Truffle especially in testing and deploying smart contracts. While they both use Javascript under the hood, however, a notably new tool called foundry not relying on Javascript was said to be more efficient. Let's see if the statement is true by building a generic staking DApp using foundry with frontend to interact with it. 
+
+# Prerequisites​
+
+Above beginner level, you should have knowledge of general programming. If you're a Rust developer, you might want to dig deeper into how to things work underneath. Basically, you do not have to worry about this. 
+
+- Have knowledge of solidity programming.
+- You should understand Typescript and/or Javascript, and
+- NextJs or ReactJs.
+
+# Requirements​
+
+This tutorial is to guide you on building a staking dApp. It will be in two parts:
+
+- Smart contracts
+- Backend/Frontend
+
```

Comment:
In the past years, Javascript has been very helpful in web3 development: compiling, testing, deploying, and interacting with smart contracts. It was however argued that Hardhat was a significant improvement over Truffle, especially in testing and deploying smart contracts. While they both use Javascript under the hood, however, a notably new tool called foundry does not rely on Javascript and was said to be more efficient. Let's see if the statement is true by building a generic staking DApp using foundry with a frontend to interact with it. 
		

		# Prerequisites​
		

		Above beginner level, you should know general programming. If you're a Rust developer, you might want to dig deeper into how things work underneath. You do not have to worry about this. 
		

		- Know solidity programming.
		- You should understand Typescript and/or Javascript, and
		- NextJs or ReactJs.